adder.go

```diff
@@ -11,6 +11,8 @@ type adder struct{}
 var adderInstance *adder
 
 func (r *adder) Direct(ctx context.Context, scopedMap map[string]interface{}, scopedAttr string, value interface{}) bool {
+	scopedMap, scopedAttr = applyDotNotation(scopedMap, scopedAttr)
+
```

Comment:
In the current design, we generally delegate value resolution and movement to `scope.go` or `scimpatch.go`, so I think it would make sense to handle dot notation there as well.

My current thought is that it might be most natural to invoke `applyDotNotation` inside `pathUnspecifiedOperate` in `scimpatch.go`, to resolve and re-map the paths accordingly.